File: SparkScala/src/main/scala/com/feng/hive/HiveTest.scala
package com.feng.hive

import org.apache.spark.sql.SparkSession

/**
 * hive查询测试
 * author: fqy
 */

object HiveTest {
  private val APP_NAME = "Hive 测试"

  def main(args: Array[String]): Unit = {
    System.setProperty("HADOOP_USER_NAME", "admin")
    println("=== 启动Spark任务 ===")
    if (args == null || args.length != 0) {
      println("invalid params! ")
      for (i <- args) {
        println("params: " + i)
      }
    } else {
      val sc = new SparkSession.Builder().appName(APP_NAME).enableHiveSupport.getOrCreate
      val res = sc.sql("SELECT * FROM xem.column_job").count()
      println("====[ " + res + " ]=====")
      println("=== 计算完成 ===")
      sc.close()
    }
  }
}

File: SparkJava/src/main/java/com/feng/HiveTest.java
package com.feng;

import org.apache.commons.lang3.StringUtils;
import org.apache.spark.sql.SaveMode;
import org.apache.spark.sql.SparkSession;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * @author fqy
 * @since 2020/11/18
 */

public class HiveTest {
    private static final String APP_NAME = "Hive 测试";
    private static final Logger logger = LoggerFactory.getLogger(HiveTest.class);

    public static void main(String[] args) {

        logger.info("=== 启动Spark任务 ===");

        // 检查参数的合法性
        if (args == null
                || args.length < 1
                || StringUtils.isBlank(args[0])) {
            logger.info("invalid params!");
        } else {
            SparkSession session = new SparkSession.Builder().appName(APP_NAME).enableHiveSupport().getOrCreate();
            session.sql("SELECT * FROM test")
                    .filter("id > " + args[0])
                    .write().format("orc").mode(SaveMode.Overwrite).saveAsTable("test1");
            logger.info("=== 计算完成 ===");
            session.close();
        }
    }
}
